modules/apps/portal-settings/portal-settings-web/src/main/java/com/liferay/portal/settings/web/internal/portlet/action/EditCompanyMVCActionCommand.java

```diff
@@ -314,9 +315,33 @@ protected void updateCompany(ActionRequest actionRequest) throws Exception {
 			industry, type, size, languageId, timeZoneId, addresses,
 			emailAddresses, phones, websites, unicodeProperties);
 
+		_updateCompanySitesLocale(companyId, unicodeProperties);
+
 		_portal.resetCDNHosts();
 	}
 
+	private void _updateCompanySitesLocale(
+		long companyId, UnicodeProperties unicodeProperties) {
+
+		List<Group> groups = _groupLocalService.getActiveGroups(
+			companyId, true);
+
+		String newCompanyLanguageIds = unicodeProperties.getProperty(
+			PropsKeys.LOCALES);
+
+		for (Group group : groups) {
+			if (group.getType() == GroupConstants.TYPE_SITE_OPEN) {
```

Comment:
Should we only update open membership groups?  What about restricted/private?  Or is this a check for something else?

Reply:
Okay, I will update this condition, to consider the other two types.

Reply:
It would be better to just check if the group is a site (or any type which can have a language specified), regardless of the membership type